Source/cmGlobalVisualStudio15Generator.cxx

```diff
@@ -172,75 +171,36 @@ bool cmGlobalVisualStudio15Generator::IsWin81SDKInstalled() const
   return false;
 }
 
-std::string const& cmGlobalVisualStudio15Generator::GetMSBuildCommand()
-{
-  if (!this->MSBuildCommandInitialized) {
-    this->MSBuildCommandInitialized = true;
-    this->MSBuildCommand = this->FindMSBuildCommand();
-  }
-
-  return this->MSBuildCommand;
-}
-
 std::string cmGlobalVisualStudio15Generator::FindMSBuildCommand()
 {
-  // For VS2017, cmake is deployed at <VSRoot>\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin
-  // Use relative path to find the MSBuild location
-  std::string cmakeCommandLocation = cmSystemTools::GetCMakeCommand();
-  std::size_t found = cmakeCommandLocation.find("/Common7/IDE/CommonExtensions/Microsoft/cmake/cmake/bin");
-  if (found != std::string::npos) {
-    std::string MSBuildPath = cmakeCommandLocation.substr(0, found);
-    MSBuildPath += "/MSBuild/15.0/Bin/MSBuild.exe";
-
-    if (cmSystemTools::FileExists(MSBuildPath.c_str()) != true) {
-      return MSBuildPath;
-    } else {
-      return "";
+  std::string msbuild;
+
+  // Ask Visual Studio Installer tool.
+  std::string vs;
+  if (vsSetupAPIHelper.GetVSInstanceInfo(vs)) {
+    msbuild = vs + "/MSBuild/15.0/Bin/MSBuild.exe";
+    if (cmSystemTools::FileExists(msbuild)) {
+      return msbuild;
     }
   }
-  else {
-    // If MSBuild path is not yet found the query the VS Setup API to get it
-    std::string VSLocation = "";
-    if (!vsSetupAPIHelper.GetVSInstanceInfo(VSLocation)) {
-      return "";
-    }
 
-    std::string MSBuildLocation =
-      VSLocation + "/MSBuild/15.0/Bin/MSBuild.exe";
-    if (cmSystemTools::FileExists(MSBuildLocation.c_str()) != true) {
-      return "";
-    }
-    return MSBuildLocation;
-  }
+  msbuild = "MSBuild.exe";
+  return msbuild;
 }
 
 std::string cmGlobalVisualStudio15Generator::FindDevEnvCommand()
 {
-  // For VS2017, cmake is deployed at <VSRoot>\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin
-  // Use relative path to find the MSBuild location
-  std::string cmakeCommandLocation = cmSystemTools::GetCMakeCommand();
-  std::size_t found = cmakeCommandLocation.find("/Common7/IDE/CommonExtensions/Microsoft/cmake/cmake/bin");
-  if (found != std::string::npos) {
-    std::string DevEnvPath = cmakeCommandLocation.substr(0, found);
-    DevEnvPath += "/Common7/IDE/devenv.exe";
-
-    if (cmSystemTools::FileExists(DevEnvPath.c_str()) != true) {
-      return DevEnvPath;
-    }
-    else {
-      return "";
+  std::string devenv;
+
+  // Ask Visual Studio Installer tool.
+  std::string vs;
+  if (vsSetupAPIHelper.GetVSInstanceInfo(vs)) {
+    devenv = vs + "/Common7/IDE/devenv.com";
```

Comment:
Devenv.com? Why did you change this part?

Reply:
Because this is what is in Kitware. Unless we have to change it it's much easier to just keep what they have.

Reply:
Okay sounds good. Thanks